### ISSUE TITLE
<list>: Consider retuning the splice family to take advantage of Ice Lake and later store pairing

### ISSUE THREAD

**Issue by BillyONeal** [microsoft/STL#166]
The September update of the Intel Architecture Optimization Manual is out: https://software.intel.com/sites/default/files/managed/9e/bc/64-ia-32-architectures-optimization-manual.pdf

New in Ice Lake is the ability for the chip to fuse multiple stores to the same cache line so that they both occur in the same cycle. (Search for "store pairing" in the document)

As a result it might make sense to rearrange the load and store sequence at https://github.com/microsoft/STL/tree/cd8fee03209d3312bf3c79abf81371290f116d84/stl/inc#L469 to do all the stores to each list node edited together.

**Comment by CaseyCarter** [comment]
> https://github.com/microsoft/STL/tree/cd8fee03209d3312bf3c79abf81371290f116d84/stl/inc#L469

Do you mean https://github.com/microsoft/STL/blob/cd8fee03209d3312bf3c79abf81371290f116d84/stl/inc/list#L469? `stl/inc` is a directory.

**Comment by BillyONeal** [comment]
Yes. I just highlighted the line, pressed y, and copied out the result, I didn't try to follow it...
